src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -765,18 +765,37 @@ void CodeGen::genSSE2Intrinsic(GenTreeHWIntrinsic* node)
             assert(op2 != nullptr);
             assert(baseType == TYP_DOUBLE);
 
+            ins    = Compiler::insOfHWIntrinsic(intrinsicID, baseType);
             op2Reg = op2->gtRegNum;
             ival   = Compiler::ivalOfHWIntrinsic(intrinsicID);
             emit->emitIns_SIMD_R_R_R_I(ins, emitTypeSize(TYP_SIMD16), targetReg, op1Reg, op2Reg, ival);
 
             break;
         }
 
+        case NI_SSE2_LoadFence:
+        case NI_SSE2_MemoryFence:
```

Comment:
nit: It seems better to just duplicate the asserts here and have the cases handled separately.

Reply:
Agree, it will be at least couple of cycles less